CustomTabBar/TabButton.swift

```diff
@@ -156,7 +164,21 @@ private extension TabButton {
             titleLabel.font = UIFont.systemFont(ofSize: 16)
         }
     }
-    
+
+    func updateBadge() {
+        guard let value = dataObject.badgeValue else { badgeLabel.removeFromSuperview(); return }
+        badgeLabel.text = value
+        badgeLabel.textAlignment = .center
+        badgeLabel.backgroundColor = .red
```

Comment:
I think we'd want to parameterize this. Different designs will probably call for slightly different shades of red with their color scheme.